.travis.yml

```diff
@@ -58,19 +45,47 @@ cache:
     - $HOME/Library/Caches/Homebrew
 
 before_install:
-  - if [[ "$CXX" == "g++" ]];
+  - if [ "$TRAVIS_OS_NAME" == "osx" ];
+    then
+       export CC=clang;
+       export CXX=clang++;
+       export GCOV=gcov;
+       export CTEST_ARGS=-VV;
+    elif [[ "$CXX" == "g++" ]];
     then
+       export CC=gcc-8;
+       export CXX=g++-8;
+       export GCOV=gcov-8
        export CTEST_ARGS=-V;
     else
+       export CC=clang-7;
+       export CXX=clang++-7;
+       export GCOV=gcov-8;
        export CTEST_ARGS=-VV;
```

Comment:
I guess lets just get rid of the explicit suffix and thus eliminate the branches as well

Reply:
On it.

Reply:
How do we handle the Ubuntu names?

Reply:
We could use `update-alternatives` inside the container:

```
sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-7 1000
sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-7 1000
...
```